File: 2502-sort-the-people/2502-sort-the-people.java
class Solution {
    public String[] sortPeople(String[] names, int[] heights) {
        TreeMap<Integer, String> mp = new TreeMap<Integer, String>();
        for (int i = 0; i < heights.length; i++) mp.put(heights[i], names[i]);
        int idx = names.length - 1;
        for (Integer h: mp.keySet()) names[idx--] = mp.get(h);
        return names;
    }
}

File: 2502-sort-the-people/2502-sort-the-people.cpp
class Solution {
public:
    vector<string> sortPeople(vector<string>& names, vector<int>& heights) {
        unordered_map<int, string> mp;
        for (int i = 0; i < heights.size(); i++) mp[heights[i]] = names[i];
        sort(heights.begin(), heights.end(), greater<int>());
        for (int i = 0; i < heights.size(); i++) names[i] = mp[heights[i]];
        return names;
    }
};

auto init = []() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    cout.tie(nullptr);
    return 'c';
}();

